[CI] LangPainlessClientYamlTestSuiteIT test {yaml=painless/146_dense_vector_bit_basic/Cosine Similarity is not supported} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3209 / oraclelinux-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/l5vuaymyrrc6m)
- [elasticsearch-periodic #3232 / openjdk21_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/y4sc4jqxiukck)
- [elasticsearch-pull-request #23907 / part-1](https://gradle-enterprise.elastic.co/s/lv7itwob3x4t6)

**Reproduction Line:**
```
./gradlew ':modules:lang-painless:yamlRestTest' --tests "org.elasticsearch.painless.LangPainlessClientYamlTestSuiteIT.test {yaml=painless/146_dense_vector_bit_basic/Cosine Similarity is not supported}" -Dtests.seed=ED89F8CD62B783B9 -Dtests.locale=es-ES -Dtests.timezone=America/Lower_Princes -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Byaml%3Dpainless%2F146_dense_vector_bit_basic%2FCosine%20Similarity%20is%20not%20supported%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.painless.LangPainlessClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [painless/146_dense_vector_bit_basic:136]: expected [400] status code but api [search] returned [200 OK] [{"took":2,"timed_out":false,"_shards":{"total":2,"successful":1,"skipped":0,"failed":1,"failures":[{"shard":0,"index":"test-index","node":"3RCBJqISShqsIk6rXEn1DA","reason":{"type":"script_exception","reason":"runtime error","script_stack":["org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.script.field.vectors.BitBinaryDenseVector.cosineSimilarity(BitBinaryDenseVector.java:71)","org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.script.VectorScoreScriptUtils$ByteCosineSimilarity.cosineSimilarity(VectorScoreScriptUtils.java:385)","org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.script.VectorScoreScriptUtils$CosineSimilarity.cosineSimilarity(VectorScoreScriptUtils.java:428)","cosineSimilarity(params.query_vector, 'vector')","                       ^---- HERE"],"script":"cosineSimilarity(params.query_vector, 'vector')","lang":"painless","position":{"offset":23,"start":0,"end":47},"caused_by":{"type":"unsupported_operation_exception","reason":"cosineSimilarity is not supported for bit vectors."}}}]},"hits":{"total":0,"max_score":null,"hits":[]}}]
```

**Issue Reasons:**
- [main] 3 failures in test test {yaml=painless/146_dense_vector_bit_basic/Cosine Similarity is not supported} (1.0% fail rate)
- [main] 1.0% failure rate in test test {yaml=painless/146_dense_vector_bit_basic/Cosine Similarity is not supported} (1.0% fail rate)

**Note:**
This issue was created using new, in-development muting automation. Please mind any issues.

--- a/modules/lang-painless/src/yamlRestTest/resources/rest-api-spec/test/painless/146_dense_vector_bit_basic.yml
+++ b/modules/lang-painless/src/yamlRestTest/resources/rest-api-spec/test/painless/146_dense_vector_bit_basic.yml
@@ -8,6 +8,8 @@ setup:
       indices.create:
         index: test-index
         body:
+          settings:
+            number_of_shards: 1
           mappings:
             properties:
               vector:
@@ -107,7 +109,6 @@ setup:
       headers:
         Content-Type: application/json
       search:
-        rest_total_hits_as_int: true
         body:
           query:
             script_score:
@@ -138,7 +139,6 @@ setup:
       headers:
         Content-Type: application/json
       search:
-        rest_total_hits_as_int: true
         body:
           query:
             script_score:
@@ -152,7 +152,6 @@ setup:
       headers:
         Content-Type: application/json
       search:
-        rest_total_hits_as_int: true
         body:
           query:
             script_score:
@@ -167,7 +166,6 @@ setup:
       headers:
         Content-Type: application/json
       search:
-        rest_total_hits_as_int: true
         body:
           query:
             script_score:

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -76,12 +76,6 @@ tests:
 - class: org.elasticsearch.compute.lucene.ValueSourceReaderTypeConversionTests
   method: testLoadAll
   issue: https://github.com/elastic/elasticsearch/issues/110244
-- class: org.elasticsearch.painless.LangPainlessClientYamlTestSuiteIT
-  method: test {yaml=painless/146_dense_vector_bit_basic/Cosine Similarity is not supported}
-  issue: https://github.com/elastic/elasticsearch/issues/110290
-- class: org.elasticsearch.painless.LangPainlessClientYamlTestSuiteIT
-  method: test {yaml=painless/146_dense_vector_bit_basic/Dot Product is not supported}
-  issue: https://github.com/elastic/elasticsearch/issues/110291
 - class: org.elasticsearch.action.search.SearchProgressActionListenerIT
   method: testSearchProgressWithQuery
   issue: https://github.com/elastic/elasticsearch/issues/109867

